server/src/main/java/org/opensearch/search/wlm/QueryGroupHelper.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.wlm;
+
+import org.opensearch.search.ResourceType;
+import org.opensearch.tasks.Task;
+
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Helper class for calculating resource usage for different resource types.
+ */
+public class QueryGroupHelper {
```

Comment:
intentionally kept the name of the class generic and not scoping to resource usage as ` `QueryGroupResourceUsageHelper`, so that this class can have more helper methods in the future